### PR TITLE
Added forms

### PR DESCRIPTION
**Pull request by GuilhermeGiachellin**
## What was done in this PR:

- Added forms to add comments, posts, and likes.
- The comments and likes are displayed in the show view of posts.
- The post is created for the current user.
- To create a new post, use the `http://localhost:3000/new` path.

image

image

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CommentsController < ApplicationController
+  def create
+    @user = User.find(params[:user_id])
```
Comment (sinansevgi) [inline app/controllers/comments_controller.rb:3]: - [x] You should create comments on behalf of the current_user as requested in the project requirements.

> Create a comment on behalf of the current_user.

```ruby
    @user = current_user
```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -8,4 +8,20 @@ def show
...
+  end
+
+  def create
+    @post = current_user.posts.new(params.require(:post).permit(:title, :text))
+    redirect_to user_post_path(current_user, @post) if @post.save
+  end
+
+  def like
+    @post = Post.all.find(params[:id])
+    @user = User.find(params[:user_id])
```
Comment (sinansevgi) [inline app/controllers/posts_controller.rb:23]: - [x] Same as the comment, you should save the like on behalf of the current_user.

For example
```ruby
current_user.likes.create(post_id: @post.id)
```

**Review by sinansevgi — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
------

Hi @GuilhermeGiachellin, 

Good job so far!
I have posted a new post without an issue :clap:
There are some issues that you still need to work on to go further but you are almost there!

 ## Required Changes ♻️
_Please check the inline comments for my suggestions_ :arrow_down:

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi

------

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (GuilhermeGiachellin) [reply app/controllers/posts_controller.rb:23]: Hello reviewer! 

Thank for the review!

These are the changes that I made:

- When I create the like, the `author_id` receives the current_user id:

image

- The same thing with the comment:

image

I just maintain `@user` to return to the correct route when the changes are made.

Again, thank you.

**Comment by GuilhermeGiachellin** [comment]
Hello reviewer! 

Thank for the review!

These are the changes that I made:

- When I create the like, the `author_id` receives the current_user id:

image

- The same thing with the comment:

image

I just maintain `@user` to return to the correct route when the changes are made.

Again, thank you.

**Review by sinansevgi — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @GuilhermeGiachellin , 

Great Work :+1:
Comments and likes getting created on behalf of current_user now!

Your project is complete!  
Congratulations! 🎉

You can merge this pull request and continue with your journey :wave: 


Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**